bin/tx-prover/src/commands/proxy.rs

```diff
@@ -23,25 +28,33 @@ impl StartProxy {
             .workers
             .iter()
             .map(|worker| format!("{}:{}", worker.host, worker.port))
-            .map(|worker| Backend::new(&worker).expect("Failed to create backend"))
-            .collect::<Vec<Backend>>();
+            .map(|worker| Backend::new(&worker).map_err(|err| err.to_string()))
+            .collect::<Result<Vec<Backend>, String>>()?;
 
-        let worker_lb = LoadBalancer::new(workers, &proxy_config);
+        let worker_lb = LoadBalancer::new(workers, &proxy_config).await;
+
+        let health_check_service = background_service("health_check", worker_lb);
+        let worker_lb = health_check_service.task();
 
         // Set up the load balancer
-        let mut lb = http_proxy_service(&server.configuration, worker_lb);
+        let mut lb = http_proxy_service(&server.configuration, LoadBalancerWrapper(worker_lb));
 
         let proxy_host = proxy_config.host;
         let proxy_port = proxy_config.port.to_string();
         lb.add_tcp(format!("{}:{}", proxy_host, proxy_port).as_str());
-        let logic = lb.app_logic_mut().expect("No app logic found");
+        let logic = lb.app_logic_mut().ok_or("Failed to get app logic")?;
         let mut http_server_options = HttpServerOptions::default();
 
         // Enable HTTP/2 for plaintext
         http_server_options.h2c = true;
         logic.server_options = Some(http_server_options);
 
+        server.add_service(health_check_service);
         server.add_service(lb);
-        server.run_forever();
+        tokio::task::spawn_blocking(|| server.run_forever())
+            .await
+            .map_err(|err| err.to_string())?;
```

Comment:
We do so because Pingora create a new runtime in `.run_forever()` without the possibility of passing and existing one. The `tokio::task::spawn_block` was introduced in order to avoid a panic each time that we instantiate a proxy.